src/pybind/mgr/cephadm/ssh.py

```diff
@@ -153,7 +160,7 @@ async def _execute_command(self,
             r = await conn.run(f'{sudo_prefix}true', check=True, timeout=5)
             r = await conn.run(cmd, input=stdin)
         # handle these Exceptions otherwise you might get a weird error like TypeError: __init__() missing 1 required positional argument: 'reason' (due to the asyncssh error interacting with raise_if_exception)
-        except (asyncssh.ChannelOpenError, asyncssh.ProcessError, Exception) as e:
```

Comment:
I don't really understand this patch. It seems we're making separate cases here that raise a `ProcessError` instead of a `HostConnectionError` but then we're handling those error types exactly the same way in the `check_host` function? Why not just have us also include the stderr in the current existing case if we think that's something users ought to see instead of making new error types?

Reply:
if they don't have passwordless sudo access the most likely case is they're trying to add a new host in which case whether we mark it offline or not is irrelevant since we aren't tracking the host anyway. 

What cases raise a process error vs. a connection error? Also, I'd still think we could just add stderr to the exception message if it's available and just not add it if it isn't rather than a full new error type just for adding stderr to the message.

Reply:
> if they don't have passwordless sudo access the most likely case is they're trying to add a new host in which case whether we mark it offline or not is irrelevant since we aren't tracking the host anyway.
> 

Good, was just wondering if it could leave to some weird situation.

> What cases raise a process error vs. a connection error? Also, I'd still think we could just add stderr to the exception message if it's available and just not add it if it isn't rather than a full new error type just for adding stderr to the message.

For example, in case you do have ssh connectivity to the remote host but you don't have sudo access. This would raise a ProcessError as you cannot execute the command. In general it would be any command that you will try to run remotely and fails with an error code (we can obtain that as well in the case of ProcessError).

I agree on having only once exception, but probably in that case we should rename it in something other than `HostConnectionError` as this name clearly indicates an issue with the connectivity which could not be the case.

Reply:
sure, I don't particularly mind the rename. It's an internal thing, so if we do the name change properly it should be totally opaque to the users and we can backport that change into quincy (where `HostConnectionError` already is). I just don't want to have multiple error types where the only difference is that when we raise one we have a slightly different error message.